.github/workflows/run_tests.yml

```diff
@@ -0,0 +1,58 @@
+name: Run Unit Tests
+
+# Remove this for now, circleci is getting in the way with tests failing
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  # trigger on pull requests
+  pull_request:
+
+  # trigger on all commits to master and tagged commits
+  push:
+    branches:
+      - "master"
+
+  # trigger on request
+  workflow_dispatch:
+
+jobs:
+  run_tests:
+    name: Run tests on ${{ matrix.os }} with python ${{ matrix.pyver-env.pyver }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macOS-latest, ubuntu-latest, windows-latest]
+        pyver-env: [ {pyver: '3.6', env: 'test-oldest_env.yaml'},
+                     {pyver: '3.7', env: 'test-compatible_env.yaml'},
+                     {pyver: '3.8', env: 'test_env.yaml'},
+                     {pyver: '3.9', env: 'test_env.yaml'},
+                     {pyver: '3.10', env: 'test_env.yaml'} ]
```

Comment:
3x5 jobs is a pretty large support matrix. We can shrink this to 3 operating systems by 4 configurations (3.8 oldest, 3.8, 3.9, 3.10).